app/models/compute_resources/foreman/model/ovirt.rb

```diff
@@ -30,6 +30,42 @@ def supports_update?
       true
     end
 
+    def supports_operating_systems?
+      client.respond_to?(:operating_systems) || client.send(:client).respond_to?(:operating_systems)
+    end
+
+    def determine_os_type(host)
+      return nil unless host
+      ret = "other_linux"
+      return ret unless host.operatingsystem
+      os_name = os_name_mapping(host)
+      arch_name = arch_name_mapping(host)
+
+      best_matches = available_operating_systems.select { |os| os.name.present? }.sort_by do |os|
```

Comment:
Instead of sorting, just keep track of the largest rating value and its associated OS - just to save some cycles